packages/lib-auth/src/client.js

```diff
@@ -0,0 +1,318 @@
+const cookie = require('cookie')
+const isNode = require('detect-node')
+
+const createImmutable = require('./helpers/create-immutable')
+
+/**
+ * The main Client class, which provides methods to authenticate against the API and request access tokens.
+ *
+ * @class
+ */
+class Client {
+  /**
+   * Creates a new instance of the client
+   *
+   * @param {object} config - Configuration parameters for the client
+   * @param {object} httpClient - A configured HTTP client for making requests
+   */
+  constructor (config, httpClient) {
+    createImmutable(this, '_config', config)
+    createImmutable(this, '_httpClient', httpClient)
+    this._resetState()
+  }
+
+  /**
+   * Makes a request to exchange a JWT for a access token.
+   *
+   * @async
+   * @private
+   * @param {string} jwt - A JSON Web Token
+   * @returns {Promise} Resolves to the new access token.
+   */
+  async _exchangeJWTForToken (jwt) {
```

Comment:
I’m confused. I thought the JWT that we use for auth is the token that’s returned from this method. JWT here seems to refer to the Panoptes session cookie.